webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -1,311 +1,111 @@
+    const [username, setUsername] = useState(user.username);
+    const [firstName, setFirstName] = useState(user.first_name);
+    const [lastName, setLastName] = useState(user.last_name);
+    const [nickname, setNickname] = useState(user.nickname);
+    const [position, setPosition] = useState(user.position);
+    const [originalEmail, setOriginalEmail] = useState(user.email);
+    const [email, setEmail] = useState('');
+    const [confirmEmail, setConfirmEmail] = useState('');
+    const [currentPassword, setCurrentPassword] = useState('');
+    const [pictureFile, setPictureFile] = useState<File>();
+    const [loadingPicture, setLoadingPicture] = useState(false);
+    const [sectionIsSaving, setSectionIsSaving] = useState(false);
+    const [clientError, setClientError] = useState('');
+    const [serverError, setServerError] = useState('');
+    const [emailError, setEmailError] = useState('');
+
+    const requireEmailVerification = useSelector((state: GlobalState) => getConfig(state).RequireEmailVerification === 'true');
+
+    const resetState = useCallback(() => {
+        setUsername(user.username);
+        setFirstName(user.first_name);
+        setLastName(user.last_name);
+        setNickname(user.nickname);
+        setPosition(user.position);
+        setOriginalEmail(user.email);
+        setEmail('');
+        setConfirmEmail('');
+        setCurrentPassword('');
+        setPictureFile(undefined);
+        setLoadingPicture(false);
+        setSectionIsSaving(false);
+        setServerError('');
+    }, [user.email, user.first_name, user.last_name, user.nickname, user.position, user.username]);
+
+    const updateSection = useCallback((section: string) => {
+        resetState();
+        setClientError('');
+        setServerError('');
+        setEmailError('');
+        setSectionIsSaving(false);
+
+        submitActive.current = false;
+        updateSectionProp(section);
+    }, [resetState, updateSectionProp]);
+
+    const submitUser = useCallback((user: UserProfile, emailUpdated: boolean) => {
+        setSectionIsSaving(true);
+
+        dispatch(updateMe(user)).
             then(({data, error: err}) => {
                 if (data) {
-                    this.updateSection('');
+                    updateSection('');
 
-                    const verificationEnabled = this.props.requireEmailVerification && emailUpdated;
+                    const verificationEnabled = requireEmailVerification && emailUpdated;
                     if (verificationEnabled) {
-                        this.props.actions.clearErrors();
-                        this.props.actions.logError({
+                        dispatch(clearErrors());
+                        dispatch(logError({
                             message: AnnouncementBarMessages.EMAIL_VERIFICATION_REQUIRED,
                             type: AnnouncementBarTypes.SUCCESS,
-                        }, true);
+                        }, true));
```

Comment:
arent we doing this in the updateMe action already?

Reply:
Not in the webapp side at least. We are logging errors when the errors come from the server, but not if the verification enabled is set.